pkg/identity/cache/allocator.go

```diff
@@ -260,7 +290,7 @@ func Release(ctx context.Context, owner IdentityAllocatorOwner, id *identity.Ide
 
 	// This will block until the kvstore can be accessed and all identities
 	// were successfully synced
-	WaitForInitialIdentities(ctx)
+	WaitForInitialGlobalIdentities(ctx)
```

Comment:
Check the error returned.

Reply:
See https://github.com/cilium/cilium/pull/8345#discussion_r295916349

Reply:
Filed this as a good first issue: https://github.com/cilium/cilium/issues/8368